### PR TITLE
feat: Expose MPSideloadedKit class to public API

### PR DESCRIPTION
**Pull request by rmi22186**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
This PR exposes the MPSideloadedKit class to the public API and refactors code to consider the changes necessary for the new class - primarily replacing `sideloadedKit.register` with `sideloadedKit.kitInstance.register`

 ## Testing Plan
 - [x] Was this tested locally? If not, explain why.
 - {explain how this has been tested, and what, if any, additional testing should be done}

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-5351

### CODE REVIEW

**Review by markvdouw — APPROVED** [review]
LGTM

File: src/sideloadedKit.ts [hunk 1]
```diff
@@ -3,9 +3,25 @@ import { IKitFilterSettings } from './configAPIClient';
 import { UnregisteredKit } from './forwarders.interfaces';
 import { EventTypeEnum, IdentityType } from './types.interfaces';
 
-/**
- */
-export class MPSideloadedKit {
+export interface IMPSideloadedKit {
+    kitInstance: UnregisteredKit;
+    filterDictionary: IKitFilterSettings;
+    addEventTypeFilter(eventType: EventTypeEnum): void;
```
Comment (alexs-mparticle) [inline src/sideloadedKit.ts:9]: Nitpick

```suggestion
    filterDictionary: IKitFilterSettings;

    addEventTypeFilter(eventType: EventTypeEnum): void;
```

File: test/jest/sideloadedKit.spec.ts [hunk 2]
```diff
@@ -86,4 +85,25 @@ describe('MPSideloadedKit', () => {
             expect(filterDictionary.userAttributeFilters).toEqual(expectedResult);
         });
     });
+
+    describe('#MPSideloadedKit Integration Tests', () => {
```
Comment (alexs-mparticle) [inline test/jest/sideloadedKit.spec.ts:89]: Move this into a separate integration test file so we can run independently.

File: test/jest/sideloadedKit.spec.ts [hunk 3]
```diff
@@ -86,4 +85,25 @@ describe('MPSideloadedKit', () => {
...
+        function makeMPSideloadedKit(n: IMPSideloadedKitConstructor) {
+            return new n(mockKitInstance);
+        }
+        it('should create a new MPSideloadedKit instance from the global mParticle object', () => {
+            const sampleSideloadedKit = makeMPSideloadedKit(MPSideloadedKit);
+
+            expect(sampleSideloadedKit).toHaveProperty('addEventAttributeFilter');
+            expect(sampleSideloadedKit).toHaveProperty('addEventTypeFilter');
+            expect(sampleSideloadedKit).toHaveProperty('addEventNameFilter');
+            expect(sampleSideloadedKit).toHaveProperty('addEventAttributeFilter');
```
Comment (alexs-mparticle) [inline test/jest/sideloadedKit.spec.ts:100]: Looks like your indents are off. Please re-run prettier.

File: test/jest/sideloadedKit.spec.ts [hunk 4]
```diff
@@ -86,4 +85,25 @@ describe('MPSideloadedKit', () => {
             expect(filterDictionary.userAttributeFilters).toEqual(expectedResult);
         });
     });
+
+    describe('#MPSideloadedKit Integration Tests', () => {
+        // https://stackoverflow.com/questions/13407036/how-does-interfaces-with-construct-signatures-work
+        function makeMPSideloadedKit(n: IMPSideloadedKitConstructor) {
```
Comment (alexs-mparticle) [inline test/jest/sideloadedKit.spec.ts:91]: Why must this be outside of the test block?

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]

File: src/forwarders.js [hunk 5]
```diff
@@ -624,7 +624,14 @@ export default function Forwarders(mpInstance, kitBlocker) {
                 // which is typed { kits: Dictionary<constructor> }.
                 // The constructors are keyed by the name of the kit.
                 config.sideloadedKits.forEach(function(sideloadedKit) {
-                    sideloadedKit.register(sideloadedKits);
+                    try {
+                        sideloadedKit.kitInstance.register(sideloadedKits);
```
[the remark this answers is not in the document]
Comment (rmi22186) [reply src/forwarders.js:628]: Made this an epic - https://mparticle-eng.atlassian.net/browse/SQDSDKS-5426

File: test/jest/sideloadedKit.spec.ts (hunk 2 above)
Comment (rmi22186) [reply test/jest/sideloadedKit.spec.ts:89]: Added sus eng - https://mparticle-eng.atlassian.net/browse/SQDSDKS-5424

**Review by alexs-mparticle — APPROVED** [review]

**Pull request merged**